casual/casual-caller-http-client/src/main/java/se/laz/casual/http/HttpClient.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+package se.laz.casual.http;
+
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import se.laz.casual.api.buffer.CasualBuffer;
+import se.laz.casual.api.buffer.CasualBufferType;
+import se.laz.casual.api.buffer.ServiceReturn;
+import se.laz.casual.api.buffer.type.ServiceBuffer;
+import se.laz.casual.api.flags.ErrorState;
+import se.laz.casual.api.flags.ServiceReturnState;
+
+import java.io.Closeable;
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import static jakarta.ws.rs.client.Entity.entity;
+
+public class HttpClient implements Closeable
+{
+    private static final Logger LOG = Logger.getLogger(HttpClient.class.getName());
+    private final Client client;
+    public HttpClient()
+    {
+        // CDI public constructor
+        this.client = ClientBuilder.newClient();
```

Comment:
```
If the bean is in @Dependent scope, then the client holds a direct reference to its instance. A new instance of the bean is bound to the life cycle of the newly created object. A bean in @Dependent scope is never shared between multiple injection points. 
```

Reply:
It should be sufficient to just inject and use a producer with produces and dispose for the Client object. No need to do anything more.
You might want to even play around with "requestscoped" on the producer if there are any thread issues with the http client, but as you just pull a target out and read the response in a one liner i don't think you will have an issue in reality here.

Reply:
The lifespan will be exactly the same according to the CDI documentation.
Sure, I can change the code but it will work *exactly* the same as it does now.

Reply:
The Jakarta http client is thread safe according to specification.

Reply:
Re producers, this is what they should be used for ( from the CDI spec):
```
A producer method acts as a source of objects to be injected, where:
• the objects to be injected are not required to be instances of beans, or
• the concrete type of the objects to be injected may vary at runtime, or
• the objects require some custom initialization that is not performed by the bean constructor
```